src/garage/tf/models/gaussian_mlp_model2.py

```diff
@@ -0,0 +1,252 @@
+                    log_std_network = mean_std_network[..., action_dim:]
+
+            else:
+                # separate MLPs for mean and std networks
+                # mean network
+                mean_network = mlp(
+                    state_input,
+                    output_dim=action_dim,
+                    hidden_sizes=self._hidden_sizes,
+                    hidden_nonlinearity=self._hidden_nonlinearity,
+                    hidden_w_init=self._hidden_w_init,
+                    hidden_b_init=self._hidden_b_init,
+                    output_nonlinearity=self._output_nonlinearity,
+                    output_w_init=self._output_w_init,
+                    output_b_init=self._output_b_init,
+                    name='mean_network',
+                    layer_normalization=self._layer_normalization)
+
+                # std network
+                if self._adaptive_std:
+                    log_std_network = mlp(
+                        state_input,
+                        output_dim=action_dim,
+                        hidden_sizes=self._std_hidden_sizes,
+                        hidden_nonlinearity=self._std_hidden_nonlinearity,
+                        hidden_w_init=self._std_hidden_w_init,
+                        hidden_b_init=self._std_hidden_b_init,
+                        output_nonlinearity=self._std_output_nonlinearity,
+                        output_w_init=self._std_output_w_init,
+                        output_b_init=tf.constant_initializer(
+                            self._init_std_param),
+                        name='log_std_network',
+                        layer_normalization=self._layer_normalization)
+                else:
+                    log_std_network = parameter(
+                        input_var=state_input,
+                        length=action_dim,
+                        initializer=tf.constant_initializer(
+                            self._init_std_param),
+                        trainable=self._learn_std,
+                        name='log_std_network')
+
+        mean_var = mean_network
+        std_param = log_std_network
+
+        with tf.compat.v1.variable_scope('std_limits'):
+            if self._min_std_param is not None:
+                std_param = tf.maximum(std_param, self._min_std_param)
+            if self._max_std_param is not None:
+                std_param = tf.minimum(std_param, self._max_std_param)
+
+        with tf.compat.v1.variable_scope('std_parameterization'):
+            # build std_var with std parameterization
+            if self._std_parameterization == 'exp':
+                log_std_var = std_param
+            else:  # we know it must be softplus here
+                log_std_var = tf.math.log(tf.math.log(1. + tf.exp(std_param)))
+
+        dist = tfp.distributions.MultivariateNormalDiag(
+            [mean_var, log_std_var])
+
+        return mean_var, log_std_var, std_param, dist
```

Comment:
why return anything but `dist`?

Reply:
do we still need the unparameterized `std_param`?

Reply:
you tell me -- does anything use it?

there's a good reason to be able to return `mean` and `log_std` in `agent_info` (so that algorithms can avoid the overhead of a feedforward to get those). anything else i don't see a point.